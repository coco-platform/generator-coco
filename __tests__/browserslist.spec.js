/**
 * @description - coco:browserslist generator
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// packages
const path = require('path');
const fs = require('fs');
const renderer = require('yeoman-test');
const assert = require('yeoman-assert');

describe('coco:browserslist', () => {
  it('should generate browserslist configuration', () => {
    const postcss = path.resolve(__dirname, '../generators/browserslist');

    return renderer.run(postcss).then(() => {
      assert.file(['.browserslistrc']);

      const configuration = fs.readFileSync('.browserslistrc', 'utf8');

      expect(configuration).toMatchSnapshot();
    });
  });
});
