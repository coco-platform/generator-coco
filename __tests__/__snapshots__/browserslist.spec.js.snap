// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coco:browserslist should generate browserslist configuration 1`] = `
"# coco defaults browserslist
>0.5% in CN
last 2 versions
Firefox ESR
not ie <= 9
"
`;
